apps/ledger-live-mobile/src/helpers/getDeviceAnimation.ts

```diff
@@ -207,6 +208,13 @@ const animations: { [modelId in DeviceModelId]: any } = {
   },
 };
 
+// export type Animations = Record<
+//   DeviceModelId,
+//   typeof animations[keyof typeof animations]
+// >;
```

Comment:
ah you tried here, i guess there were unknown depth in the Record ...

Reply:
Ah i think i got something:

```TS
type Animation = AnimatedLottieViewProps["source"];
type AnimationRecord = { [key: string]: string | Animation | AnimationRecord };
export type Animations = { [modelId in DeviceModelId]: AnimationRecord };

const animations: Animations = { etc.. }
```

PS: i removed the "s" on your `Animations` type

Reply:
I think if we want to really type this correctly we need to go further, because even with the types you provided the `@ts-expect-error` will still be needed.

The correct types would be something like this:

```ts
type Animation = AnimatedLottieViewProps["source"];
type AnimationRecord = Record<"light" | "dark", Animation>;
type S_SP_BLUE =
  | DeviceModelId.nanoS
  | DeviceModelId.nanoSP
  | DeviceModelId.blue;
export type Animations = {
  [modelId in S_SP_BLUE]: Record<string, AnimationRecord>;
} & {
  [modelId in DeviceModelId]: Record<string, Record<string, AnimationRecord>>;
};
```

I'll make the change.

Reply:
should be good now 😄